builds/jazz-build-module/sls-app/sbr.groovy

```diff
@@ -139,40 +145,391 @@ class IntValidator implements TypeValidator {
+    try {
+      if(!aValue instanceof JSONObject)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class SequenceValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      if(!aValue instanceof List)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class IamArnValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^arn:aws:iam::\\d{12}:role/?[a-zA-Z_0-9+=,.@\\-_/]+"
+      def match = aValue ==~ pattern
+      if(!match)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class KmsArnValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^arn:aws:kms::\\d{12}:key/?[a-zA-Z_0-9+=,.@\\-_/]+"
+      def match = aValue ==~ pattern
+      if(!match)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class AwsIdValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^\\d{12}"
+      def match = aValue ==~ pattern
+      if(!match)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class FunctionValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^[a-zA-Z_0-9+=,.@\\-_/]+"
+      def match = aValue ==~ pattern
+      if(!match)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class ResourceValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^[a-zA-Z]+"
+      def match = aValue ==~ pattern
+      if(!match)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class PolicyValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^[a-zA-Z]+"
+      def match = aValue ==~ pattern
+      if(!match)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class EventValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^[a-zA-Z]+"
+      def match = aValue ==~ pattern
+      if(!match)
+      throw new IllegalArgumentException(aValue);
+    } catch(e) {
+      throw new IllegalArgumentException(aValue);
+    }
+  }
+}
+
+class AwsVariableNameValidator implements TypeValidator {
+  public void isValid(def aValue) {
+    try {
+      def pattern = "^[a-zA-Z]+"
```

Comment:
All the four classes above are completely the same shall we need four of then then?

Reply:
I kept it separately bcoz, we can add more condition in feature if required